### ISSUE TITLE
feat: Allow custom/angular navigation on swipe to go back

### ISSUE THREAD

**Issue by jonasheinrich-akelius** [ionic-team/ionic#18316]
# Feature Request

**Ionic version:**
[x] **4.12.0**

**Describe the Feature Request**
The swipe to go back feature is tightly coupled to the `NavController`. I guess [this is the culprit](https://github.com/ionic-team/ionic/blob/b8d4961483d4add345af74d8a2e5a0a539deabfe/angular/src/directives/navigation/ion-router-outlet.ts#L47).
Since ionic 4 it seems to be the preferred way to directly use angular's router for navigation (see e.g. [here](https://ionicframework.com/docs/navigation/angular)). So it would be nice to have a bit more control over this navigation event.

**Describe Preferred Solution**
Maybe one could pass the `swipeHandler` on the directive. Something like:
```
// template
<ion-router-outlet [swipeHandler]="getSwipeHandler()"></ion-router-outlet>

// controller
getSwipeHandler() {
  return {
    canStart: () => { .. },
    onStart: () => { .. },
    onEnd: () => { .. }
  };
}
```
One downside of this approach would be that one would have to take care of animations oneself.

**Describe Alternatives**
One could also disable this feature, expose `swiper` and implement this feature oneself. But this seems a bit cumbersome. 

**Related Code**
See Describe Preferred Solution

**Additional Context**
This is primarily relevant for iOS.

**Comment by liamdebeasi** [comment]
Hi there,

Thanks for the feature request. Can you explain your use case a bit more? Are you interested in customizing the swipe to go back animation, or are you interested in customizing the behavior on swipe to go back?

Thanks!

**Comment by jonasheinrich-akelius** [comment]
Hi Liam, 

we created a rather simple navigation service our own which can simply navigate to urls and go back via `location.back`. So no history stack or the like. Maybe this is naive but so far it worked good. :)

But ionic keeps track of its own history stack. So we had the following bug:
1. Start at `/`.
2. Go to `/A`.
3. Go to `/B`.
4. Swipe back -> `/A`.
5. Go back -> `/B`.

But we actually want to be at `/` after 5. because we went back two times. So the problem is that ionic has its own stack of pages and this conflicts with any other stack/navigation implementation. 
I just realized that ionic may need this stack for its page caching algorithm. Maybe one would have to disable this as well.. :(

Did this help? Thanks in advance!

Jonas
